x-pack/legacy/plugins/siem/public/pages/detection_engine/rules/index.tsx

```diff
@@ -61,7 +65,11 @@ export const RulesComponent = React.memo(() => {
             </EuiFlexItem>
 
             <EuiFlexItem grow={false}>
-              <EuiButton fill href="#/detection-engine/rules/create" iconType="plusInCircle">
+              <EuiButton
+                fill
+                href={`#${DETECTION_ENGINE_PAGE_NAME}/rules/create`}
```

Comment:
Consider using the route helpers (which appear to be out of date already... 😅 ):

https://github.com/elastic/kibana/blob/59e0a1cba191e8075ad8b711b33d3378a0fd531c/x-pack/legacy/plugins/siem/public/components/link_to/redirect_to_detection_engine.tsx#L48

Reply:
When we fix the state url, we will go back to it